File: report/src/test/java/com/hartwig/actin/report/interpretation/EvaluatedCohortTestFactory.java
package com.hartwig.actin.report.interpretation;

import org.apache.logging.log4j.util.Strings;
import org.jetbrains.annotations.NotNull;

public final class EvaluatedCohortTestFactory {

    private EvaluatedCohortTestFactory() {
    }

    @NotNull
    public static ImmutableEvaluatedCohort.Builder builder() {
        return ImmutableEvaluatedCohort.builder()
                .trialId(Strings.EMPTY)
                .acronym(Strings.EMPTY)
                .isPotentiallyEligible(false)
                .isOpen(false)
                .hasSlotsAvailable(false);
    }
}

File: report/src/test/kotlin/com/hartwig/actin/report/interpretation/EvaluatedCohortTestFactory.kt
package com.hartwig.actin.report.interpretation

object EvaluatedCohortTestFactory {

    fun evaluatedCohort(
        trialId: String = "",
        acronym: String = "",
        isPotentiallyEligible: Boolean = false,
        isOpen: Boolean = false,
        hasSlotsAvailable: Boolean = false,
        molecularEvents: Iterable<String> = emptySet(),
        cohort: String? = null
    ): EvaluatedCohort {
        return EvaluatedCohort(
            trialId = trialId,
            acronym = acronym,
            cohort = cohort,
            isPotentiallyEligible = isPotentiallyEligible,
            isMissingGenesForSufficientEvaluation = false,
            isOpen = isOpen,
            hasSlotsAvailable = hasSlotsAvailable,
            molecularEvents = molecularEvents.toSet(),
            warnings = emptySet(),
            fails = emptySet()
        )
    }
}